addons/sale/migrations/11.0.1.1/post-migration.py

```diff
@@ -8,9 +9,77 @@
+def fill_sale_order_line_computed(env):
+    """Compute through SQL sale.order.line fields for speeding up migration."""
+    # We first set everything to 0 in amt_invoiced
+    openupgrade.logged_query(
+        env.cr, "UPDATE sale_order_line sol SET amt_invoiced = 0.00",
+    )
+    # Then we set the quantity for the order lines with related invoice lines
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET amt_invoiced = sub.amount
+        FROM (
+            SELECT rel.order_line_id, SUM(ail.price_total) AS amount
+            FROM sale_order_line_invoice_rel rel
+                INNER JOIN account_invoice_line ail ON
+                    rel.invoice_line_id = ail.id
+                INNER JOIN account_invoice ai ON ai.id = ail.invoice_id
+            WHERE ai.state IN ('open', 'paid')
+            GROUP BY rel.order_line_id
+        ) AS sub
+        WHERE sol.id = sub.order_line_id""",
+    )
+    # Now we set the amt_invoiced with this simplified computation, but before
+    # discounting refunded invoiced. We apply a fixed rounding for avoiding
+    # resolution problems. It also uses a trick using the same table 2 times
+    # for avoiding to perform a subquery.
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET amt_to_invoice = ROUND(
+            (CASE WHEN COALESCE(pt.invoice_policy, 'x') = 'delivery'
+             THEN sol.price_reduce_taxinc * sol.qty_delivered
+             ELSE sol.price_total - sol.amt_invoiced END), 5
+        )
+        FROM sale_order_line sol2
+        LEFT JOIN product_product pp ON pp.id = sol2.product_id
+        LEFT JOIN product_template pt ON pt.id = pp.product_tmpl_id
+        WHERE sol2.id = sol.id""",
+    )
+    # We have to do this second query that reduces the invoiced amount due to
+    # the refunded amount as the computation mixes several tables that can't
+    # be combined together - The computation is not exact, even on upstream,
+    # as there's no real link - See https://github.com/odoo/odoo/pull/26054
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET amt_invoiced = sol.amt_invoiced - sub.amount
+        FROM (
+            SELECT sol.id,
+                SUM(ai2.amount_total) AS amount
+            FROM sale_order_line sol
+                INNER JOIN sale_order_line_invoice_rel rel
+                    ON sol.id = rel.order_line_id
+                INNER JOIN account_invoice_line ail
+                    ON rel.invoice_line_id = ail.id
+                INNER JOIN account_invoice ai ON ai.id = ail.invoice_id
+                INNER JOIN account_invoice ai2 ON ai2.refund_invoice_id = ai.id
+            WHERE ai.state IN ('open', 'paid')
+                AND ai2.state IN ('open', 'paid')
+                AND ail.product_id = sol.product_id
+            GROUP BY sol.id
+        ) AS sub
+        WHERE sol.id = sub.id""",
```

Comment:
```
UPDATE sale_order_line sol
SET amt_invoiced = sol.amt_invoiced - sub.amount
FROM (
	SELECT rel.order_line_id, SUM(ail.price_total) AS amount
	FROM sale_order_line_invoice_rel rel
	LEFT JOIN sale_order_line sol2 ON sol.id = rel.order_line_id
	INNER JOIN account_invoice_line ail ON
	    rel.invoice_line_id = ail.id
	INNER JOIN account_invoice ai ON ai.id = ail.invoice_id
        INNER JOIN account_invoice ai2 ON ai2.refund_invoice_id = ai.id
    WHERE ai.state IN ('open', 'paid')
	AND ai2.state IN ('open', 'paid')
	AND ail.product_id = sol2.product_id
    GROUP BY rel.order_line_id
) AS sub
WHERE sol.id = sub.order_line_id
```

Reply:
I don't remember why I put you this query. It seems both do the same 😅

Reply:
Hehe, let's forget it then!